doc/plugins.dox

```diff
@@ -186,59 +195,143 @@ Let's start with the very basic headers and add more later if necessary.
 
 Now you can go on and write your first lines for your new plugin. As mentioned before,
-you will need to implement and fill out a couple of functions/macros to make the plugin work.
-So let's start with PLUGIN_VERSION_CHECK().
-
-PLUGIN_VERSION_CHECK() is a convenient way to tell Geany which version of Geany's plugin API
-is needed at minimum to run your plugin. The value is defined in
+you will need to implement and a couple of functions. Arguably the most important
+one is @a geany_load_module(). This is the function that makes Geany see you plugin at all. When
+Geany scans the pre-defined and user-configured plugin directories, it will take a look at each
+shared library (or DLL on Windows) to see if it exports a @a geany_load_module() symbol. Files
+lacking these will be outright ignored.
+
+@subsection register Registering a Plugin
+
+Geany will then invoke this function, always and regardless of whether the user activates your
+plugin. In fact its purpose to probe if the plugin should even be presented to the user. Therefore
+it important that you use this function to register your plugin, making it available to the user.
+Geany will pass a pointer to a GeanyPlugin instance which acts as a unique handle to your plugin.
+Use this pointer for registering and later API calls. Registering the plugin consists of a number
+of steps:
+
+  1. Filling GeanyPlugin::info with metadata
+    - @ref PluginInfo::name : The name of your plugin
+    - @ref PluginInfo::description : A brief description.
+    - @ref PluginInfo::version : The current version.
+    - @ref PluginInfo::author : Your contanct information.
+    .
+    Filling all of them is recommended to provide the best user experience, but only the name is truly
+    mandatory.
+  2. Filling GeanyPlugin::funcs with function pointers
+    - @ref GeanyPluginFuncs::init : an initialization function
+    - @ref GeanyPluginFuncs::cleanup : a finalization function
+    - @ref GeanyPluginFuncs::configure : a function that provides configuration (optional)
+    - @ref GeanyPluginFuncs::help : a function that provides help (optional)
+    - @ref GeanyPluginFuncs::callbacks : a pointer to an array of PluginCallback (optional).
+    .
+    @a init and @a cleanup are mandatory, the other ones depend on how advanced your plugin is.
+    Furthermore, the @a init and @a cleanup
+    pair is called when the user activates your plugin in the Plugin Manager. So use these to
+    do advanced initialization and teardown as to not waste resources when the plugin is not even
+    enabled.
+  3. Registering for real by calling @a geany_plugin_register(). You must pass the GeanyPlugin
+    pointer that your received and filled out as above. geany_plugin_register() also takes
+    various version numbers which are detailed below. Please <b>use the macro @a
+    GEANY_PLUGIN_REGISTER()</b> where you only specify the minimum API version. Please also
```

Comment:
"Please use the wrapper macro GEANY_PLUGIN_REGISTER() where you only specify the minimum API version instead of directly calling geany_plugin_register()."

BTW, it's nice to speak about `geany_plugin_register()` but maybe a bit confusing here as you then say not to use it.